### ISSUE TITLE
codegen `TimestampWithTimeZone` support

### ISSUE THREAD

**Issue by tyt2y3** [SeaQL/sea-orm#170]
Hello,

I have kind of the same problem but need orientation to solve that.

## First things first

1. My postgres table has a column called created_at with type "timestamp with time zone"
2. I use sea-orm-cli to generate entity (set DATABASE_URL environment variable then sea-orm-cli generate entity -o some_dir)
3. I notice that the columns generated have the NaiveDateTime type (I see pub created_at: DateTime, declared on the generated model).

## Environment

* sea-cli-orm v 2.0.0
* In my project, I have sea-orm = { version = "^0", features = ["sqlx-postgres", "runtime-tokio-native-tls", "macros"], default-features = false }

## Error

Queries run show the following error:

mismatched types; Rust type `core::option::Option<chrono::naive::datetime::NaiveDateTime>` (as SQL type `TIMESTAMP`) is not compatible with SQL type `TIMESTAMPTZ`"

If I try to use DateTime<UTC> (by changing the generated entities by hand) I have a trait error.

Without waiting for the cli to correctly recognize the type (if it isn't still doing so), how can I fix that?

Thanks,
Marlon

_Originally posted by @marlon-sousa in https://github.com/SeaQL/sea-orm/discussions/68#discussioncomment-1343661_

-----

We might have a bug in codegen? Or is it in SeaSchema?

**Comment by billy1624** [comment]
Hey @marlon-sousa, thanks for the report!

What should be correct corresponding Rust type of Postgres`timestamp with time zone`?
- I guess it's `chrono::DateTime<Utc>` / `chrono::DateTime<Local>` and perhaps `chrono::DateTime<chrono::FixedOffset>`?
- References
   - https://docs.rs/sqlx/0.5.7/sqlx/postgres/types/index.html#chrono
   - https://docs.rs/sqlx/0.5.7/sqlx/struct.Postgres.html#impl-Decode%3C%27r%2C%20Postgres%3E-6

This line might be at fault, it's inside `sea-orm-codegen`
- https://github.com/SeaQL/sea-orm/blob/38a4d3a810e8ccf1567266646de140edc4291a21/sea-orm-codegen/src/entity/column.rs#L40

**Comment by marlon-sousa** [comment]
Hello,

I mapped this by hand in the generated entity and it worked.

I however have another question though: Other orm's map this kind of timestamp to ``` Chrono::DateTime<Utc> ``` instead of ``` chrono::DateTime<FixedOffset> ```

I have yet to analize what the difference is, but if you ave any insight about why we should use one ratter than the other this would be great.

**Comment by tyt2y3** [comment]
It should have been fixed in `0.2.2`.

`DateTime<Utc>` is a special case of `DateTime<FixedOffset>` where `offset` = 0.
The rationale is, timestamps should be in UTC.
In fact, there isn't a concept of `timestamp with timezone` in MySQL, so the more intuitive mapping is `timestamp` to `NaiveDateTime`, which implies that it is in UTC.

**Comment by marlon-sousa** [comment]
Hello,

I performed a cargo install-update -a and sea-orm-cli were updated from 0.2.0 to 0.2.2

In fact, the generated entities are now using the correct DateWithTimezone, but something else is broken because the code no longer compiles.

I am letting you know because may be 0.2.1 broke compatibility and sea-orm isn't yet released so anyone trying to make this migration may get trouble.

Errors are below

```
error[E0432]: unresolved import `super::account_routes::Entity`
 --> src/database/entities/prelude.rs:3:9
  |
3 | pub use super::account_routes::Entity as AccountRoutes;
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `Entity` in `database::entities::account_routes`

```



It seems that 0.2.2 is not generating a Entity struct for each table. 0.2.0 generates

```
#[derive(Copy, Clone, Default, Debug, DeriveEntity)]
pub struct Entity;
```

whereas 0.2.2 does not. Strangely enough, the prelud from 0.2.2 still references the Entity struct that 0.2.2 itself no longer generates.

```
error: cannot find derive macro `DeriveEntityModel` in this scope
  --> src/database/entities/account_routes.rs:5:35
   |
5  | #[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
   |                                   ^^^^^^^^^^^^^^^^^ help: a derive macro with a similar name exists: `DeriveActiveModel`
   | 
  ::: /home/marlon/.cargo/registry/src/github.com-1ecc6299db9ec823/sea-orm-macros-0.2.2/src/lib.rs:78:1
   |
78 | pub fn derive_active_model(input: TokenStream) -> TokenStream {
   | ------------------------------------------------------------- similarly named derive macro `DeriveActiveModel` defined here

error: cannot find attribute `sea_orm` in this scope
 --> src/database/entities/account_routes.rs:6:3
  |
6 | #[sea_orm(table_name = "account_routes")]
  |   ^^^^^^^

error: cannot find attribute `sea_orm` in this scope
 --> src/database/entities/account_routes.rs:8:7
  |
8 |     #[sea_orm(primary_key)]
  |       ^^^^^^^

```

It seems though that some macros aren't being found. I wounder whether this has any relation with the lack of the Entity struct reported in the first error, may be it is now being auto generated by a macro which is not included ...

I want to thank you for the great work and am reporting this in the hope that my reports can help the project to grow in a sustainable way.

Thanks,
Marlon

**Comment by tyt2y3** [comment]
Sorry about that. We were in the process of a release. Please also update `sea-orm` to `0.2.2`, which is released literally 1 minute ago.
We pair the `sea-orm` with a corresponding `sea-orm-cli` version.

**Comment by marlon-sousa** [comment]
Now it is working indeed as expected.

Feel free to close this issue.

**Comment by tyt2y3** [comment]
Thank you for your early support!

**Comment by Oodachi** [comment]
![image](https://user-images.githubusercontent.com/34698754/133941950-a7fc7230-4d5e-4b56-ade5-a2684c9965a8.png)

**Comment by billy1624** [comment]
Hey @Oodachi, thanks for the catch! This was fixed on #178.
